rules/TypeDeclaration/TypeAnalyzer/GenericClassStringTypeNormalizer.php

```diff
@@ -5,24 +5,41 @@
 namespace Rector\TypeDeclaration\TypeAnalyzer;
 
 use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\Type\ArrayType;
 use PHPStan\Type\Constant\ConstantStringType;
 use PHPStan\Type\Generic\GenericClassStringType;
 use PHPStan\Type\ObjectType;
 use PHPStan\Type\Type;
 use PHPStan\Type\TypeTraverser;
+use PHPStan\Type\UnionType;
+use Rector\Core\Configuration\Option;
+use Rector\StaticTypeMapper\ValueObject\Type\FullyQualifiedObjectType;
+use Symplify\PackageBuilder\Parameter\ParameterProvider;
 
 final class GenericClassStringTypeNormalizer
 {
     public function __construct(
-        private ReflectionProvider $reflectionProvider
+        private ReflectionProvider $reflectionProvider,
+        private ParameterProvider $parameterProvider
     ) {
     }
 
     public function normalize(Type $type): Type
     {
-        return TypeTraverser::map($type, function (Type $type, $callback): Type {
+        $isAutoImport = $this->parameterProvider->provideParameter(Option::AUTO_IMPORT_NAMES);
```

Comment:
There should also type-specific replacements that might help static analysis:

```suggestion
        $isAutoImport = $this->parameterProvider->provideBoolParameter(Option::AUTO_IMPORT_NAMES);
```

Reply:
updated 👍